frontend/lib/src/hooks/useSubmitFormViaEnterKey.test.ts

```diff
@@ -0,0 +1,245 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { waitFor } from "@testing-library/react"
+import { renderHook } from "@testing-library/react-hooks"
+
+import { WidgetStateManager } from "@streamlit/lib/src/WidgetStateManager"
+
+import useSubmitFormViaEnterKey from "./useSubmitFormViaEnterKey"
+
+const widgetMgr = new WidgetStateManager({
+  sendRerunBackMsg: vi.fn(),
+  formsDataChanged: vi.fn(),
+})
+describe("useSubmitFormViaEnterKey", () => {
+  it("should call commitWidgetValue and submitForm when Enter is pressed", async () => {
+    const commitWidgetValue = vi.fn()
+    const allowFormEnterToSubmit = vi.fn((_: string) => true)
+    const submitForm = vi.fn()
+    widgetMgr.allowFormEnterToSubmit = allowFormEnterToSubmit
+    widgetMgr.submitForm = submitForm
+
+    const { result: onInputChange } = renderHook(() =>
+      useSubmitFormViaEnterKey(
+        "someFormId",
+        commitWidgetValue,
+        true,
+        widgetMgr,
+        "someFragmentId",
+        false
+      )
+    )
+
+    onInputChange.current({
+      key: "Enter",
+      metaKey: false,
+      ctrlKey: false,
+      keyCode: 0,
+    })
+    await waitFor(() => {
+      expect(commitWidgetValue).toHaveBeenCalled()
```

Comment:
**suggestion (non-blocking):** A stricter check to ensure we aren't accidentally doing things multiple times unexpectedly (for example, if there was a bug in our hook logic) is to utilize the `toHaveBeenCallTimes(number)` assertion instead

```suggestion
      expect(commitWidgetValue).toHaveBeenCalledTimes(1)
```